packages/arb-avm-cpp/data_storage/src/arbcore.cpp

```diff
@@ -382,8 +382,7 @@ rocksdb::Status ArbCore::reorgToMessageCountOrBefore(
             std::vector<unsigned char> checkpoint_vector(
                 checkpoint_it->value().data(),
                 checkpoint_it->value().data() + checkpoint_it->value().size());
-            auto checkpoint =
-                extractMachineStateKeys(checkpoint_vector.begin());
+            auto checkpoint = extractMachineStateKeys(checkpoint_vector);
```

Comment:
As-is this would fail if the checkpoint wasn't filled it. With the return type change, this would nicely pop out as a place where we could just continue to the next checkpoint if we get one that's just a `MachineOutput`

Reply:
:+1: